.circleci/config.yml

```diff
@@ -33,10 +33,51 @@ jobs:
           bucket: mozilla-desktop-funnel-prototype
           source: /tmp/workspace/public
           options: -r
+  deploy-preview:
+    docker:
+      - image: google/cloud-sdk
+    working_directory: /tmp
+    steps:
+      - attach_workspace:
+          # Must be absolute path or relative path from working_directory
+          at: /tmp/workspace
+      - gcs/gcs-rsync:
+          bucket: mozilla-desktop-funnel-prototype
+          source: /tmp/workspace/public
+          destination: "preview/$CIRCLE_BRANCH"
+          options: -r
+      # shamelessly cribbed from https://circleci.com/developer/orbs/orb/samsalisbury/github-status
+      - run: |
+          set -euo pipefail
+          CONTEXT="circleci/$CIRCLE_JOB"
+          # Compose our URL and auth pieces
+          API=https://api.github.com
+          USER=$CIRCLE_PROJECT_USERNAME
+          REPO=$CIRCLE_PROJECT_REPONAME
+          SHA=$CIRCLE_SHA1
+          AUTH=$GITHUB_STATUS_USER:$GITHUB_STATUS_TOKEN
```

Comment:
I'm currently just using a personal access token here, which is probably fine for this case, but it's not ideal. Wonder if there's better alternatives.

Reply:
I have used [deploy keys](https://docs.github.com/en/free-pro-team@latest/developers/overview/managing-deploy-keys#deploy-keys) to grant repo-based access rather than user-based access, but deploy keys are for ssh only, and I don't think can be used for the github rest api. Instead, we would need to use a machine user to make this more robust. It would be worth checking with data sre about whether we have ever deployed a machine user account for this sort of workflow; I don't think we have.